csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.shared.cs

```diff
@@ -766,43 +777,49 @@ public ulong ProfilingStartTimeNs
         #region private methods
 
         private void Init(string modelPath, SessionOptions options,
-                          PrePackedWeightsContainer prepackedWeightsContainer = null)
+                          PrePackedWeightsContainer prepackedWeightsContainer = null, IntPtr? envHandle = null)
```

Comment:
It would be better to pass the env object. We confine native handles to the code that actually invokes the native API.